File: nastroje/vytvor-spravce.py
"""
Spuštění:
  pip install firebase-admin
  python nastroje/vytvor-spravce.py cesta/k/spravci.csv cesta/k/serviceAccount.json
"""

import sys
import csv
import time
import firebase_admin
from firebase_admin import credentials, auth

def main():
    if len(sys.argv) < 3:
        print("Použití: python vytvor-spravce.py spravci.csv serviceAccount.json")
        sys.exit(1)

    csv_soubor = sys.argv[1]
    sa_soubor = sys.argv[2]

    cred = credentials.Certificate(sa_soubor)
    firebase_admin.initialize_app(cred)

    spravci = []
    with open(csv_soubor, encoding="utf-8-sig") as f:
        reader = csv.DictReader(f, delimiter=";")
        for radek in reader:
            id_ = radek.get("ID", "").strip()
            heslo = radek.get("Heslo", "").strip()
            jmeno = radek.get("Jmeno", radek.get("Jméno", "")).strip()
            prijmeni = radek.get("Prijmeni", radek.get("Příjmení", "")).strip()
            if id_ and heslo:
                spravci.append({
                    "id": id_,
                    "heslo": heslo,
                    "jmeno": f"{jmeno} {prijmeni}".strip()
                })

    print(f"Načteno {len(spravci)} správců")
    ok = preskoceno = chyby = 0

    for s in spravci:
        email = f"{s['id']}@mojebudky.cz"
        try:
            auth.create_user(email=email, password=s["heslo"], display_name=s["jmeno"])
            print(f"✓ {s['id']} – {s['jmeno']}")
            ok += 1
        except firebase_admin.auth.EmailAlreadyExistsError:
            print(f"– {s['id']} již existuje")
            preskoceno += 1
        except Exception as e:
            print(f"✗ {s['id']}: {e}")
            chyby += 1

    print(f"\nHotovo! Vytvořeno: {ok}, přeskočeno: {preskoceno}, chyby: {chyby}")

if __name__ == "__main__":
    main()

File: nastroje/vytvor-spravce.js
// Spusť: node vytvor-spravce.js
// Potřebuješ: npm install firebase-admin
// a soubor serviceAccount.json ve stejné složce

const admin = require("firebase-admin");
const serviceAccount = require("./serviceAccount.json");

admin.initializeApp({
  credential: admin.credential.cert(serviceAccount),
});

const spravci = [
  { id: "001400", heslo: "INYOW7", jmeno: "Aja" },
  { id: "002852", heslo: "lHZ4e7", jmeno: "Květa" },
  { id: "003",    heslo: "dtLj7E", jmeno: "Pavel" },
  { id: "004112", heslo: "Hq8Koa", jmeno: "" },
  { id: "005484", heslo: "pWftq3", jmeno: "Vali + Viki" },
  { id: "006187", heslo: "8WEP6j", jmeno: "Honza" },
  { id: "007109", heslo: "KFYtjX", jmeno: "Jirka" },
  { id: "008666", heslo: "2th0zq", jmeno: "Karel" },
  { id: "009112", heslo: "sfnPAz", jmeno: "Plíšek" },
  { id: "010488", heslo: "FMjR0c", jmeno: "Jirka" },
  { id: "011506", heslo: "kBall7", jmeno: "Štěpánek" },
];

async function vytvorUcty() {
  for (const s of spravci) {
    const email = `${s.id}@mojebudky.cz`;
    try {
      await admin.auth().createUser({
        email,
        password: s.heslo,
        displayName: s.jmeno,
      });
      console.log(`✓ ${s.id} (${s.jmeno})`);
    } catch (err) {
      if (err.code === "auth/email-already-exists") {
        console.log(`- ${s.id} již existuje`);
      } else {
        console.error(`✗ ${s.id}: ${err.message}`);
      }
    }
  }
  console.log("Hotovo!");
  process.exit(0);
}

vytvorUcty();
